test/check-application

```diff
@@ -22,46 +25,132 @@ registries = ['localhost:5000']
+            if (items[i].innerText.trim().startsWith(first) && items[i].innerText.trim().endsWith(last))
+                return true;
+        return false;
+        })""", name, owner)
+
 class TestApplication(testlib.MachineCase):
 
     def setUp(self):
         # backup/restore pristine podman state, so that tests can run on existing testbeds
         # HACK: sometimes podman leaks mounts
         super().setUp()
         self.machine.execute("cp -a /var/lib/containers /var/lib/containers.orig")
+
+        # Create admin session
+        self.machine.execute("mkdir /home/admin/.ssh")
+        self.machine.execute("cp /root/.ssh/* /home/admin/.ssh")
+        self.machine.execute("chown -R admin:admin /home/admin/.ssh")
+        self.machine.execute("chmod 700 /home/admin/.ssh")
+        self.machine.execute("chmod 600 /home/admin/.ssh/*")
+        self.admin_s = ssh_connection.SSHConnection(user="admin",
+                                               address=self.machine.ssh_address,
+                                               ssh_port=self.machine.ssh_port,
+                                               identity_file=self.machine.identity_file)
+
+        # Enable user service as well
+        if self.machine.image != "rhel-8-1":
+            self.admin_s.execute("systemctl --now --user enable io.podman.socket")
+        else:
+            self.allow_journal_messages("Unit io.podman.socket could not be found.")
+
+        self.allow_journal_messages("/run.*/podman/io.podman: couldn't connect.*")
         self.addCleanup(self.machine.execute, "podman rm --force --all && "
             "findmnt --list -otarget | grep /var/lib/containers | xargs -r umount && "
             "rm -r /var/lib/containers && "
             "mv /var/lib/containers.orig /var/lib/containers")
 
-    def testBasic(self):
+    def execute(self, auth, cmd):
+        if auth:
+            return self.machine.execute(cmd)
+        else:
+            return self.admin_s.execute(cmd)
+
+    def testBasicRoot(self):
+        self._testBasic(True)
+
+    @skipImage("No user service", "rhel-8-1")
+    def testBasicUser(self):
+        self._testBasic(False)
+
+    def _testBasic(self, auth):
         b = self.browser
         m = self.machine
 
-        self.login_and_go("/podman")
+        if not auth:
+            self.allow_authorize_journal_messages()
+            self.allow_browser_errors("Failed to start root io.podman.socket.*")
+
+        self.login_and_go("/podman", authorized=auth)
         b.wait_present("#app")
+
         b.wait_present(".content-filter input")
-        b.wait_in_text("#containers-images", "busybox:latest")
+
+        # Check all containers
+        if auth:
+            checkImage(b, "docker.io/library/busybox:latest", "system")
+            checkImage(b, "docker.io/library/alpine:latest", "system")
+            checkImage(b, "docker.io/library/registry:2", "system")
+
+        if m.image != "rhel-8-1":
```

Comment:
This test case has a lot of these special cases, which makes it rather hard to read and understand. How about you split this into testSystemContainers and testUserContainers, and `skipImage` the latter on rhel-8-1?

Reply:
I want to check two scenarios:
1. Only user containers (there is `testBasicUser` which is skipped on rhel-8-1)
2. Test that user and system containers can both be used at once (this is this scenario. Problem is that rhel-8-1 does not have user containers so I need to skip half of the test).

I could copy the tests once over and have only root containers, which would be tested on rhel-8-1. Do you think I should do that?